Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -231,110 +208,276 @@ def ev02₂ {V : SSet.Truncated 2} (φ : V _⦋2⦌₂) : ev0₂ φ ⟶ ev2₂ 
+    (h₁ : ∀ (x : V _⦋0⦌₂), F.obj (mk x) = G.obj (mk x))
+    (h₂ : ∀ ⦃x y : V _⦋0⦌₂⦄ (e : Edge x y),
+      F.map (homMk e) = eqToHom (h₁ x) ≫ G.map (homMk e) ≫ eqToHom (h₁ y).symm) :
+    F = G :=
+  Functor.ext_of_iso (mkNatIso (fun x ↦ eqToIso (h₁ x))
+    (fun _ _ e ↦ by simp [h₂ e])) (fun _ ↦ h₁ _) (fun x ↦ by
+      obtain ⟨x, rfl⟩ := x.mk_surjective
+      simp)
+
+end
+
+end HomotopyCategory
+
+section
+
+open HomotopyCategory
+
+variable {V W} (f : V ⟶ W)
 
 /-- A map of 2-truncated simplicial sets induces a functor between homotopy categories. -/
-def mapHomotopyCategory {V W : SSet.Truncated.{u} 2} (F : V ⟶ W) :
+def mapHomotopyCategory :
     V.HomotopyCategory ⥤ W.HomotopyCategory :=
   CategoryTheory.Quotient.lift _
-    ((oneTruncation₂ ⋙ Cat.freeRefl).map F ⋙ HomotopyCategory.quotientFunctor W)
-    (by
-      rintro _ _ _ _ ⟨φ⟩
-      apply CategoryTheory.Quotient.sound
-      apply HoRel₂.mk' (φ := F.app _ φ)
-        (f₀₁ := (oneTruncation₂.map F).map (ev01₂ φ))
-        (f₀₂ := (oneTruncation₂.map F).map (ev02₂ φ))
-        (f₁₂ := (oneTruncation₂.map F).map (ev12₂ φ))
-      all_goals
-        apply FunctorToTypes.naturality)
+    ((oneTruncation₂ ⋙ Cat.freeRefl).map f ⋙ quotientFunctor W) (by
+      rintro _ _ _ _ ⟨h⟩
+      exact CategoryTheory.Quotient.sound _ ⟨h.map f⟩)
+
+@[simp]
+lemma mapHomotopyCategory_obj (x : V _⦋0⦌₂) :
+    (mapHomotopyCategory f).obj (.mk x) = .mk (f.app _ x) := rfl
+
+@[simp]
+lemma mapHomotopyCategory_homMk {x y : V _⦋0⦌₂} (e : Edge x y) :
+    (mapHomotopyCategory f).map (homMk e) = homMk (e.map f) := rfl
+
+end
 
 /-- The functor that takes a 2-truncated simplicial set to its homotopy category. -/
 def hoFunctor₂ : SSet.Truncated.{u} 2 ⥤ Cat.{u,u} where
-  obj V := Cat.of (V.HomotopyCategory)
-  map {S T} F := mapHomotopyCategory F
-  map_id S := by
-    apply Quotient.lift_unique'
-    simp [mapHomotopyCategory, Quotient.lift_spec]
-    exact Eq.trans (Functor.id_comp ..) (Functor.comp_id _).symm
-  map_comp {S T U} F G := by
-    apply Quotient.lift_unique'
-    simp [mapHomotopyCategory, SSet.Truncated.HomotopyCategory.quotientFunctor]
-    rw [Quotient.lift_spec, Cat.comp_eq_comp, Cat.comp_eq_comp, ← Functor.assoc, Functor.assoc,
-      Quotient.lift_spec, Functor.assoc, Quotient.lift_spec]
+  obj V := Cat.of V.HomotopyCategory
+  map F := mapHomotopyCategory F
+  map_id _ := HomotopyCategory.functor_ext (by simp) (by cat_disch)
+  map_comp _ _ := HomotopyCategory.functor_ext (by simp) (by cat_disch)
 
 theorem hoFunctor₂_naturality {X Y : SSet.Truncated.{u} 2} (f : X ⟶ Y) :
     (oneTruncation₂ ⋙ Cat.freeRefl).map f ⋙ SSet.Truncated.HomotopyCategory.quotientFunctor Y =
       SSet.Truncated.HomotopyCategory.quotientFunctor X ⋙ mapHomotopyCategory f := rfl
 
+/-- By `Quotient.lift_unique'` (not `Quotient.lift`) we have that `quotientFunctor V` is an
+epimorphism. -/
+theorem HomotopyCategory.lift_unique' (V : SSet.Truncated.{u} 2) {D} [Category D]
```

Comment:
```suggestion
theorem HomotopyCategory.lift_unique' (V : SSet.Truncated.{u} 2) {D: Type*} [Category D]
```
Let’s take the occasion to do this. (I won’t nit about the universe parameters because hopefully `Category*` will land soon.)